homeassistant/components/mqtt/climate.py

```diff
@@ -603,15 +604,21 @@ async def _set_climate_attribute(
         await self._publish(cmnd_topic, payload)
         return changed
 
-    @abstractmethod
     async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set hvac mode."""
+        raise NotImplementedError()
+
+    async def async_set_operation_mode(self, operation_mode: str) -> None:
+        """Set operation mode."""
+        raise NotImplementedError()
 
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperatures."""
-        operation_mode: HVACMode | None
+        operation_mode: HVACMode | str | None
         if (operation_mode := kwargs.get(ATTR_HVAC_MODE)) is not None:
             await self.async_set_hvac_mode(operation_mode)
+        if (operation_mode := kwargs.get(ATTR_OPERATION_MODE)) is not None:
+            await self.async_set_operation_mode(operation_mode)
```

Comment:
This seems a bit odd.
Maybe the concrete classes should override `async_set_temperature` with the check for `ATTR_HVAC_MODE` / `ATTR_OPERATION_MODE` and then call `super().async_set_temperature()`?

Reply:
I think the intention is to keep as much functionality as possible in the `MqttTemperatureControlEntity` to keep the implementation classes slim [as discussed here](https://github.com/home-assistant/core/pull/93644#pullrequestreview-1464413014). Either way would work, but keeping platform specific code out of the shared class felt better to me

Reply:
We should only move the specific code out:

```
        if (operation_mode := kwargs.get(ATTR_OPERATION_MODE)) is not None:
            await self.async_set_operation_mode(operation_mode)
```
And call the super class for the shared part which should be marked as abstract method.
Then we can remove the specific methods we now added to the shared class.

Reply:
@hookedonunix can you rebase and update move the specific code the same way as done for mqtt climate?